driver/pparse.ml

```diff
@@ -195,8 +210,6 @@ let parse_all ~tool_name parse_fun magic ppf sourcefile =
   ast
 
 let parse_implementation ppf ~tool_name sourcefile =
-  parse_all ~tool_name Parse.implementation
-    Config.ast_impl_magic_number ppf sourcefile
+  parse_file ~tool_name Structure ppf sourcefile
 let parse_interface ppf ~tool_name sourcefile =
-  parse_all ~tool_name Parse.interface
-    Config.ast_intf_magic_number ppf sourcefile
+  parse_file ~tool_name Signature ppf sourcefile
```

Comment:
Since the types of `parse_{implementation,interface}` are not fixed by hand anymore, would it not make sense to expose a generic `parse` in order to replace `parse_implementation` and `parse_interface` with `parse Structure` and `parse Signature` respectively?